.github/workflows/eden.yml

```diff
@@ -60,8 +60,50 @@ jobs:
           echo "${{ env.workflow_files_modified }}"
           echo "workflow_files_modified=${{ env.workflow_files_modified }}" >> $GITHUB_OUTPUT
 
+  check_existing_workflow:
+    if: github.event.review.state == 'approved'
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.check_status.outputs.should_run }}
+    steps:
+      - name: Check if workflow is currently running
+        id: check_status
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          REPO="${{ github.repository }}"
+          PR_SHA="${{ github.event.pull_request.head.sha }}"
+
+          # Fetch running workflows for this PR
+          RUNNING_WORKFLOWS=$(curl -s -L \
+            -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "Accept: application/vnd.github+json" \
+            "https://api.github.com/repos/$REPO/actions/runs?head_sha=$PR_SHA&status=in_progress")
```

Comment:
Does including the PR_SHA mean that if there was an update to the PR it will run? The description doesn't state this which is why I'm asking.